### PR TITLE
Feature/add classes

### PR DESCRIPTION
**Pull request by Felipe-Perez-Ferraro**
- Created the person class
- Created the student class
- Created the teacher class

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @Felipe-Perez-Ferraro,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project that does not have a readme documentation.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @id = rand(1..100)
+    @age = age
+    @name = name
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    puts @parent_permission || of_age?
+  end
```
Comment (shahadat3669) [inline person.rb]: Hey there! First of all, great job on adding the `can_use_services` method. However, based on the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=Public%20method%20can_use_services%3F%20that%20returns%20true%20if%20person%20is%20of%20age%20or%20if%20they%20have%20permission%20from%20parents.), this method should return either true or false. Currently, it seems to be returning nothing. So, please make sure to update your code to meet the project requirements.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (shahadat3669) [inline person.rb:20]: Well, your `of_age` method is working fine. However, as per project requirements, this method should be private. Please update your code and make it private as per project requirements.

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @Felipe-Perez-Ferraro, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ Student class look fine 👍 
✅ Teacher class look fine 👍 
✅ Good Readme file


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Strangeal — APPROVED** [review]
Hi @Felipe-Perez-Ferraro, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
-  Fixed `can_use_services?` method to return boolean  ✅ 
- Made `of_age?` method private ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**